### PR TITLE
Closes #3297: Adds `array_equal`

### LINKED ISSUE #3297: array_equal
Implement an `array_equal` function to mirror numpy:
https://numpy.org/doc/stable/reference/generated/numpy.array_equal.html

### PR DESCRIPTION
**Pull request by drculhane**
I've tried to address all of the comments, and update/rebase this to the current master.

Closes #3297

### CODE REVIEW

File: arkouda/numeric.py [hunk 1]
```diff
@@ -2201,6 +2203,51 @@ def count_nonzero(pda):
...
+    --------
+    >>> a = ak.randint(0,10,10,dtype=ak.float64)
+    >>> b = a
+    >>> ak.array_equal(a,b)
+    True
+    >>> b[9] = np.nan
+    >>> ak.array_equal(a,b)
+    False
+    >>> a[9] = np.nan
+    >>> array_equal(a,b)
```
Comment (ajpotts) [inline arkouda/numeric.py:2240]: Should be `ak.array_equal(a,b)` (instead of `array_equal(a,b)`) to match the others.

File: arkouda/numeric.py [hunk 2]
```diff
@@ -2201,6 +2203,51 @@ def count_nonzero(pda):
...
+    Returns
+    -------
+    boolean
+      With string data: False if one array is of type ak.str_ and the other isn't,
+                        True if both are ak.str_ and they match.
+      With numeric data: True if neither array has any nan elements, and all elements pairwise equal.
+                         False if equal_Nan is False, and either array has any nan element.
+                         True if equal_Nan is True, all non-nan elements are pairwise equal,
+                                            and all nans in pda_a correspond to nans in pda_b
+
```
Comment (ajpotts) [inline arkouda/numeric.py:2229]: Still one formatting issue in the doc strings.  See the gray highighted portion of the screen shot:

Screenshot from 2024-08-30 15-28-18

**Review by ajpotts — APPROVED** [review]
Just the two minor doc string issues.  

Great work!  Thank you.

**Review by stress-tess — APPROVED** [review]
looks great! thanks andy!!

**Review by jaketrookman — APPROVED** [review]
Looks good to me

**Pull request merged**